src/analyzer.rs

```diff
@@ -262,9 +270,13 @@ fn extract_container(
             "integer" => extract_integer_type(value)?,
             "array" => {
                 // recurse through repeated arrays until we find a concrete type (keep track of how deep we went)
-                let (array_type, recurse_level) = array_recurse_for_type(value, stack, key, 1)?;
+                let (mut array_type, recurse_level) = array_recurse_for_type(value, stack, key, 1)?;
                 trace!("got array {} for {} in level {}", array_type, key, recurse_level);
-                array_recurse_level.insert(key.clone(), recurse_level);
+                if !cfg.no_condition && key == "conditions" && is_conditions(value) {
```

Comment:
just confirming, do we want the new behavior to be the default?

Reply:
yeah, i just made the bools default false to match common conventions around deriving `Default`